# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1319629495 14400
#      Wed Oct 26 07:44:55 2011 -0400
# Node ID 8cd08124cb5900122bdfc378b3990428e90cc4fa
# Parent  e652ff4d15225c06da1c3ea833a0a76dd9d906b8
Fix sloppy code and bugs introduced by changeset bda7b080f205.

diff --git a/scripts/plot/private/__go_draw_axes__.m b/scripts/plot/private/__go_draw_axes__.m
--- a/scripts/plot/private/__go_draw_axes__.m
+++ b/scripts/plot/private/__go_draw_axes__.m
@@ -342,44 +342,38 @@ function __go_draw_axes__ (h, plot_strea
     ylogscale = strcmpi (axis_obj.yscale, "log");
     zlogscale = strcmpi (axis_obj.zscale, "log");
 
     ## Detect logscale and negative lims
     if (xlogscale && all (axis_obj.xlim < 0))
       axis_obj.xsgn = -1;
       if (strcmp (axis_obj.xdir, "reverse"))
         axis_obj.xdir = "normal";
-      else
-        axis_obj.xdir = "reverse";
       endif
       axis_obj.xtick = -flip (axis_obj.xtick);
       axis_obj.xticklabel = flip (axis_obj.xticklabel);
       axis_obj.xlim = -flip (axis_obj.xlim);
     else
       axis_obj.xsgn = 1;
     endif
     if (ylogscale && all (axis_obj.ylim < 0))
       axis_obj.ysgn = -1;
       if (strcmp (axis_obj.ydir, "reverse"))
         axis_obj.ydir = "normal";
-      else
-        axis_obj.ydir = "reverse";
       endif
       axis_obj.ytick = -flip (axis_obj.ytick);
       axis_obj.yticklabel = flip (axis_obj.yticklabel);
       axis_obj.ylim = -flip (axis_obj.ylim);
     else
       axis_obj.ysgn = 1;
     endif
     if (zlogscale && all (axis_obj.zlim < 0))
       axis_obj.zsgn = -1;
       if (strcmp (axis_obj.zdir, "reverse"))
         axis_obj.zdir = "normal";
-      else
-        axis_obj.zdir = "reverse";
       endif
       axis_obj.ztick = -flip (axis_obj.ztick);
       axis_obj.zticklabel = flip (axis_obj.zticklabel);
       axis_obj.zlim = -flip (axis_obj.zlim);
     else
       axis_obj.zsgn = 1;
     endif
 
@@ -1725,17 +1719,17 @@ function __go_draw_axes__ (h, plot_strea
   endif
 
 endfunction
 
 function x = flip (x)
   if (size (x, 1) == 1)
     x = fliplr (x);
   elseif (size (x, 2) == 1 || ischar (x))
-    x = flipup (x);
+    x = flipud (x);
   else
     x = flipud (fliplr (x));
   endif
 endfunction
 
 function fontspec = create_fontspec (f, s, gp_term)
   if (strcmp (f, "*") || strcmp (gp_term, "tikz"))
     fontspec = sprintf ("font \",%d\"", s);
